proxy/http2/HPACK.cc

```diff
@@ -430,7 +438,13 @@ HpackDynamicTable::_evict_overflowed_entries()
     (*h)->value_get(&value_len);
 
     this->_current_size -= ADDITIONAL_OCTETS + name_len + value_len;
-    this->_mhdr->field_delete(*h, false);
+
+    if (this->_mhdr_old && this->_mhdr_old->fields_count() != 0) {
+      this->_mhdr_old->field_delete(*h, false);
+    } else {
+      this->_mhdr->field_delete(*h, false);
+    }
+
```

Comment:
Reviewing the GC code, I don't understand how you know whether to delete from _mhdr or _mhdr_old.

Reply:
Dynamic Table is FIFO, so when we have both `_mhdr` & `_mhdr_old`, we add a new entry to the `_mhdr` and remove from the `_mhdr_old` (until the `_mhdr_old` becomes empty).

Reply:
@shinrich Kindly ping. I can explain more if needed.

Reply:
@masaori335 Could you elaborate on 
> we add a new entry to the _mhdr and remove from the _mhdr_old 

I didn't find any reference to  _mhdr_old in `HpackDynamicTable::add_header_field`. Could you help me to understand what I missed? Thanks!

Reply:
`add_header_field()` calls `_evict_overflowed_entries()` and come to this block.